extensions/cdc-debezium/src/main/java/com/hazelcast/jet/cdc/impl/HazelcastListDatabaseHistory.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2020 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.cdc.impl;
+
+import com.hazelcast.collection.IList;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.instance.impl.HazelcastInstanceFactory;
+import io.debezium.config.Configuration;
+import io.debezium.document.Document;
+import io.debezium.document.DocumentReader;
+import io.debezium.document.DocumentWriter;
+import io.debezium.relational.history.AbstractDatabaseHistory;
+import io.debezium.relational.history.DatabaseHistoryException;
+import io.debezium.relational.history.DatabaseHistoryListener;
+import io.debezium.relational.history.HistoryRecord;
+import io.debezium.relational.history.HistoryRecordComparator;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+/**
+ * Database history implementation backed by Hazelcast IList {@link IList}.
+ *
+ * @since 4.2
+ */
+public class HazelcastListDatabaseHistory extends AbstractDatabaseHistory {
+
+    private volatile String instanceName;
+    private volatile String listName;
+
+    private volatile IList<byte[]> list;
```

Comment:
does these need to be volatile? probably not?

Reply:
Is read and written by different threads and not safely published. But anyways, let's see if we can move this whole thing into the snapshot.

Reply:
Have made the schema history part of the snapshot. There are two relatively shady parts in the change, I would be thankful if you would take a look at those:

1. see how I use `DatabaseHistoryImpl.container().setHistory(...)` to pass the right record list to `DatabaseHistoryImpl`; `DatabaseHistoryImpl` gets initilized by the surrounded `SourceTask.start(...)` and this happens on a blocking worker thread

2. in `CdcSource.State` I've used a `CopyOnWriteArrayList` for the schema history; the reasoning here was that items get added to this list on some internal debezium snapshot thread, while restore and our snapshotting reads it on a blocking worker thread, so I need to ensure proper visibility + the list is written only on schema changes, which should be a rare event after the initial db snapshot so the garbage created should be a one-time hit; I think it's correct, but doesn't feel right so I'm open to suggestions...